pkg/resources/poddistribution/poddistribution.go

```diff
@@ -0,0 +1,386 @@
+	ant := obj.GetAnnotations()
+	// If there are no annotations then 0 balance attempts have been made
+	if ant == nil {
+		return true
+	}
+	if val, ok := ant[PodBalanceAttempts]; ok {
+		i, err := strconv.Atoi(val)
+		if err != nil {
+			logrus.Errorf("Error converting string annotations %s", ant[PodBalanceAttempts])
+			return false
+		} else {
+			if i >= maxBalanceAttempts {
+				logrus.Warningf("Reached max balance attempts for %s on %s", obj.GetName(), obj.GetNamespace())
+				return false
+			}
+		}
+	}
+
+	return true
+}
+
+func getObject(ctx context.Context, client k8sclient.Client, objDtls map[string]string) (metav1.Object, error) {
+	if objDtls[kind] == "DeploymentConfig" {
+		return getDeploymentConfig(ctx, client, objDtls)
+	}
+	if objDtls[kind] == "ReplicaSet" {
+		return getReplicaSet(ctx, client, objDtls)
+	}
+	if objDtls[kind] == "StatefulSet" {
+		return getStatefulSet(ctx, client, objDtls)
+	}
+	logrus.Errorf("Unexpected obj kind %s", objDtls[kind])
+	return nil, nil
+}
+
+func getDeploymentConfig(ctx context.Context, client k8sclient.Client, objDtls map[string]string) (metav1.Object, error) {
+	deploymentConfig := &appsv1.DeploymentConfig{}
+	err := client.Get(ctx, k8sTypes.NamespacedName{Name: objDtls[name], Namespace: objDtls[ns]}, deploymentConfig)
+	if err != nil {
+		logrus.Errorf("Error getting DeploymentConfig %v", err)
+		return nil, err
+	}
+	return deploymentConfig, nil
+}
+
+func getReplicaSet(ctx context.Context, client k8sclient.Client, objDtls map[string]string) (metav1.Object, error) {
+	rs := &k8appsv1.ReplicaSet{}
+	err := client.Get(ctx, k8sTypes.NamespacedName{Name: objDtls[name], Namespace: objDtls[ns]}, rs)
+	if err != nil {
+		logrus.Errorf("Error getting ReplicaSet %v", err)
+		return nil, err
+	}
+	return rs, nil
+}
+
+func getStatefulSet(ctx context.Context, client k8sclient.Client, objDtls map[string]string) (metav1.Object, error) {
+	ss := &k8appsv1.StatefulSet{}
+	err := client.Get(ctx, k8sTypes.NamespacedName{Name: objDtls[name], Namespace: objDtls[ns]}, ss)
+	if err != nil {
+		logrus.Errorf("Error getting StatefulSet %v", err)
+		return nil, err
+	}
+	return ss, nil
+}
+
+func parseKnn(knn string) map[string]string {
+	objDtls := map[string]string{}
+	s := strings.Split(knn, "/")
+	objDtls[kind] = s[0]
+	objDtls[ns] = s[1]
+	objDtls[name] = s[2]
+	return objDtls
+}
+
+func getNamespaces(nsPrefix string, installType string) []string {
+	namespaces := []string{
+		nsPrefix + "3scale",
+		nsPrefix + "rhsso",
+		nsPrefix + "user-sso",
+	}
+
+	if installType == string(integreatlyv1alpha1.InstallationTypeManagedApi) {
+		namespaces = append(namespaces, nsPrefix+"marin3r")
+	}
+	return namespaces
+}
+
+func ReconcilePodDistribution(ctx context.Context, client k8sclient.Client, nsPrefix string, installType string) {
+	for _, ns := range getNamespaces(nsPrefix, installType) {
+		logrus.Infof("Reconciling Pod Balance in ns %s", ns)
+		unbalanced, err := findUnbalanced(ctx, ns, client)
+		if err != nil {
```

Comment:
I guess we would want to use a multi error here add a ``` continue ``` and remove the else block

Reply:
done